flink-table/flink-sql-parser/pom.xml

```diff
@@ -298,6 +348,23 @@ under the License.
 							<outputDirectory>${project.build.directory}/generated-sources/</outputDirectory>
 						</configuration>
 					</execution>
+					<execution>
+						<phase>generate-sources</phase>
+						<id>javacc-hive</id>
+						<goals>
+							<goal>javacc</goal>
+						</goals>
+						<configuration>
+							<sourceDirectory>${project.build.directory}/generated-sources-hive/</sourceDirectory>
```

Comment:
The plugin is a mess, can we just create another module instead put the code in one ? And why we copy the parser file of Flink, it seems there is no any reuse parse code block.

Reply:
We're not copying the parser file of Flink. The process of generating the parser for Hive is:
1. Make a copy of the parser template in `calcite-core.jar`
2. Generate the fmpp sources from `codegen-hive` and output to `generated-sources-hive`.
3. Call javacc to build the parser from `generated-sources-hive` and output to `generated-sources`.

Finally, we'll have both Flink and Hive parser in `generated-sources`.

I agree moving the Hive parser to a separate module may make things a little clearer. But it also brings redundancies. E.g. need extra pom file and duplicated classes like `ParseException`. What do you think?